File: API/Section/Inheritance/VB.NET/Inheritance/rptInheritBase.vb
Imports System.IO
Imports System.Resources

Public Class rptInheritBase
	Inherits GrapeCity.ActiveReports.SectionReport

	'Path to the csv file.
	Private _csvPath As String
	'Stream to load the csv file.
	Private _invoiceFileStream As StreamReader
	'String array to store the data.
	Private _fieldNameArray As String()

	Private _resource As New ResourceManager(GetType(rptInheritBase))

	Public Sub New()
		' This call is required by the ActiveReports designer.
	End Sub

	'ActiveReport overrides dispose to clean up for the list of components.
	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
		End If
		MyBase.Dispose(disposing)
	End Sub

	Protected WriteOnly Property CsvPath() As String
		Set(ByVal Value As String)
			'CsvPath property
			_csvPath = Value
		End Set
	End Property

	Protected Sub BaseReport_DataInitialize(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.DataInitialize

		'  Load CSV to stream.
		Dim invoiceFileStream As New StreamReader(_csvPath, System.Text.Encoding.GetEncoding(Convert.ToInt32(_resource.GetString("CodePage"))))


		'Read a line from the stream, and create an array string.
		Dim _currentLine As String = invoiceFileStream.ReadLine()
		_fieldNameArray = _currentLine.Split(New Char() {","c})

		'Field only to create a number of the array.
		Dim i As Integer
		For i = 0 To _fieldNameArray.Length - 1
			Fields.Add(_fieldNameArray(i).ToString())
		Next i
	End Sub

	Protected Sub BaseReport_FetchData(ByVal sender As Object, ByVal eArgs As GrapeCity.ActiveReports.SectionReport.FetchEventArgs) Handles MyBase.FetchData
		Try
			'Read a line from the stream, and creats an array string.
			Dim _currentLine As String = _invoiceFileStream.ReadLine()
			Dim _currentArray As String() = _currentLine.Split(New Char() {","c})

			'Store the Value property of Field object number of the array.
			Dim i As Integer
			For i = 0 To _currentArray.Length - 1
				Fields(_fieldNameArray(i).ToString()).Value = _currentArray(i).ToString()
			Next i

			'Set EOF to false and continue to read the data.
			eArgs.EOF = False
		Catch
			'Close the stream when the it has exceeded the time to read the last line.
			_invoiceFileStream.Close()

			'Set EOF to true and quit reading the data.
			eArgs.EOF = True
		End Try
	End Sub

	Protected Sub rptInheritBase_ReportStart(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.ReportStart
		'  Load CSV to stream.
		_invoiceFileStream = New StreamReader(_csvPath, System.Text.Encoding.GetEncoding(Convert.ToInt32(_resource.GetString("CodePage"))))
		_invoiceFileStream.ReadLine()
	End Sub
End Class

File: API/Section/Inheritance/C#/Inheritance/rptInheritBase.cs
using System;
using System.IO;
using System.Resources;

namespace GrapeCity.ActiveReports.Sample.Inheritance
{
	
	/// Base class that implements the ability to load the csv file.
	public partial class rptInheritBase : GrapeCity.ActiveReports.SectionReport
	{
		//Path to the csv file.
		private String _csvPath;

		//Stream to load the csv file.
		private StreamReader _invoiceFileStream;

		//String array to store the data.
		private string [] _fieldNameArray;

		ResourceManager _resource;
		
		public rptInheritBase()
		{
			_resource = new ResourceManager(typeof(rptInheritBase));
			// 
			// TODO:Please add a constructor logic here
			//

			//Adds an event handler.
			DataInitialize += new EventHandler(BaseReport_DataInitialize);
			FetchData += new FetchEventHandler(BaseReport_FetchData);
			ReportStart += new System.EventHandler(this.rptInheritBase_ReportStart);
		}

		//CsvPath property
		protected string CsvPath
		{
			set
			{
				_csvPath = value;
			}
		}

		protected void BaseReport_DataInitialize(object sender, System.EventArgs eArgs)
		{

			// Load CSV to stream.
			StreamReader invoiceFileStream = new StreamReader(_csvPath, System.Text.Encoding.GetEncoding(Convert.ToInt32(_resource.GetString("CodePage"))));
			//Read a line from the stream, and create an array string.
			string currentLine = invoiceFileStream.ReadLine();
			_fieldNameArray = currentLine.Split(new char[] { ',' });

			//Field only to create a number of the array.
			for (int i = 0; i < _fieldNameArray.Length; i++)
				Fields.Add(_fieldNameArray[i]);
		   
		}

		protected void BaseReport_FetchData(object sender, FetchEventArgs eArgs)
		{
			try
			{
				if (_invoiceFileStream.Peek() >= 0)
				{
					//Read a line from the stream, and creats an array string.
					string _currentLine = _invoiceFileStream.ReadLine();
					string[] _currentArray = _currentLine.Split(new char[] { ',' });

					//Store the Value property of Field object number of the array.
					for (int i = 0; i < _currentArray.Length; i++)
						Fields[_fieldNameArray[i]].Value = _currentArray[i];

					//Set EOF to false and continue to read the data.
					eArgs.EOF = false;
				}
				else
				{
					_invoiceFileStream.Close();
					eArgs.EOF = true;
				}
			}
			catch
			{
				//Close the stream when the it has exceeded the time to read the last line.
				_invoiceFileStream.Close();

				//Set EOF to true and quit reading the data.
				eArgs.EOF = true;
			}
		}

		protected void rptInheritBase_ReportStart(object sender, EventArgs e)
		{
			// Load CSV to stream.
			_invoiceFileStream = new StreamReader(_csvPath, System.Text.Encoding.GetEncoding(Convert.ToInt32(_resource.GetString("CodePage"))));
			_invoiceFileStream.ReadLine();
		}
	}
}
